doc/nrf/gs_modifying.rst

```diff
@@ -95,15 +119,67 @@ Typically, the :file:`CMakeLists.txt` files include at least the :file:`main.c`
 Configuring your application
 ****************************
 
-If your application uses a provided library or targets a specific board, you might want to change the default configuration of the library or board.
+You might want to change the default options of the application.
 There are different ways of doing this, but not all will store your configuration permanently.
 
-The default configuration for a library is specified in its :file:`Kconfig` file.
-Similarly, the default configuration for a board is specified in its :file:`*_defconfig` file (and its :file:`Kconfig.defconfig` file, see :ref:`zephyr:default_board_configuration` in the Zephyr documentation for more information).
-The configuration for your application, which might override some default options of the libraries or the board, is specified in a :file:`prj.conf` file in the application directory.
+The configuration system in |NCS| consists of several elements:
+
+* ``Kconfig`` configuration files focusing on software related options,
+* ``dts`` configuration files focusing on hardware related options,
+* ``Partition manager`` configuration files focusing on memory layout.
+
+The Zephyr and so the |NCS| configuration system is distributed.
+It means there are multiple sources of the configuration.
+
+The software related configuration comes from ``Kconfig`` files.
+There are several layers providing the configuration options:
+
+* The lowest layer results of defaults provided in ``Kconfig`` files of all system components, libraries and application itself.
+* The defaults are overlayed by the options specified by the selected board in its :file:`*_defconfig` file and its :file:`Kconfig.defconfig` file (see :ref:`zephyr:default_board_configuration` in the Zephyr documentation for more information).
+* The configuration is additionally overlayed by the :file:`prj.conf` file provided by the application.
```

Comment:
Should we also mention optional configuration overlays, specified with `-DOVERLAY_CONFIG` CMake variable?

Reply:
I was thinking about it but I would add it as a separate point to not make the matter even fuzzier. That would go as something other then build types. We need to explain them for sure as some applications use it (like asset tracker, @jhn-nordic ).

Reply:
Ok, fine with me!